smarts/core/scenario.py

```diff
@@ -180,13 +172,13 @@ def variations_for_all_scenario_roots(
             roll_agent_missions = 0
             roll_social_agents = 0
             roll_traffic_histories = 0
-
             if shuffle_scenarios:
-                roll_routes = random.randint(0, len(routes))
-                roll_agent_missions = random.randint(0, len(agent_missions))
-                roll_social_agents = random.randint(0, len(social_agents))
+                roll_routes = random.randint(1, len(routes))
+                roll_agent_missions = random.randint(1, len(agent_missions))
+                roll_social_agents = random.randint(
+                    1, len(social_agent_infos_and_missions)
+                )
```

Comment:
why was the roll needed to be at least 1?

Reply:
This is just to prevent overlap, `random.randint` gives an inclusive range `[m:n]`.  The beginning and end of the range then overlaps when using cycle.

```python
>>> rs = [random.randint(0, 3) for _ in range(10000)]
>>> min(rs) == 0
True
>>> max(rs) == 3
True
>>> list(np.roll([4,3,2], 3))
[4,3,2]
```